### ISSUE TITLE
Where scope not working for select without max, linked to a custom fieldset

### ISSUE THREAD

**Issue by erikbml** [statamic/cms#5333]
### Bug description

I have a collection: Trainer

Which has a custom fieldtype Locations (extends Relationship fieldtype), without a max (so you can link a trainer to multiple locations)

When displaying the trainers on the page, I want to have them paginated, but also filterable. For that I created a scope:

`{{ collection:trainers limit="24" paginate="true" as="instructors" query_scope="trainer_for_location" }}`

The filter is based on a text, but the value is stored as integers, so the scope is to handle the translation:

```
$location = Location::where('title', request()->location)->first();

$query->where('locations', $location->id);
```

But running this query, gives the error:

`strtolower(): Argument #1 ($string) must be of type string, array given`

@ vendor/statamic/cms/src/Query/Builder.php:343

Tried doing other stuff as well; 

`$query->where('locations:id', $location->id);`

Which doesn't give out an error, but doesn't give results either

### How to reproduce

See the bug description

### Logs

```shell
[previous exception] [object] (TypeError(code: 0): strtolower(): Argument #1 ($string) must be of type string, array given at /var/www/mib2/vendor/statamic/cms/src/Query/Builder.php:343)
[stacktrace]
#0 /var/www/mib2/vendor/statamic/cms/src/Query/Builder.php(343): strtolower()
#1 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/Builder.php(137): Statamic\\Query\\Builder->filterTestEquals()
#2 [internal function]: Statamic\\Stache\\Query\\Builder->Statamic\\Stache\\Query\\{closure}()
#3 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Collections/Arr.php(717): array_filter()
#4 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(358): Illuminate\\Support\\Arr::where()
#5 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/Builder.php(138): Illuminate\\Support\\Collection->filter()
#6 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/EntryQueryBuilder.php(95): Statamic\\Stache\\Query\\Builder->filterWhereBasic()
#7 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/EntryQueryBuilder.php(81): Statamic\\Stache\\Query\\EntryQueryBuilder->getKeysFromCollectionsWithWhere()
#8 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php(767): Statamic\\Stache\\Query\\EntryQueryBuilder->Statamic\\Stache\\Query\\{closure}()
#9 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/EntryQueryBuilder.php(84): Illuminate\\Support\\Collection->reduce()
#10 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/EntryQueryBuilder.php(62): Statamic\\Stache\\Query\\EntryQueryBuilder->getKeysFromCollectionsWithWheres()
#11 /var/www/mib2/vendor/statamic/cms/src/Stache/Query/Builder.php(124): Statamic\\Stache\\Query\\EntryQueryBuilder->getFilteredKeys()
#12 /var/www/mib2/vendor/statamic/cms/src/Query/Builder.php(313): Statamic\\Stache\\Query\\Builder->getCountForPagination()
#13 /var/www/mib2/vendor/statamic/cms/src/Tags/Concerns/GetsQueryResults.php(41): Statamic\\Query\\Builder->paginate()
#14 /var/www/mib2/vendor/statamic/cms/src/Tags/Concerns/GetsQueryResults.php(14): Statamic\\Tags\\Collection\\Entries->paginatedResults()
#15 /var/www/mib2/vendor/statamic/cms/src/Tags/Collection/Entries.php(50): Statamic\\Tags\\Collection\\Entries->results()
#16 /var/www/mib2/vendor/statamic/cms/src/Tags/Collection/Collection.php(23): Statamic\\Tags\\Collection\\Entries->get()
#17 [internal function]: Statamic\\Tags\\Collection\\Collection->__call()
#18 /var/www/mib2/vendor/statamic/cms/src/View/Antlers/Engine.php(159): call_user_func()
#19 [internal function]: Statamic\\View\\Antlers\\Engine::renderTag()
#20 /var/www/mib2/vendor/statamic/cms/src/View/Antlers/Parser.php(578): call_user_func_array()
#21 /var/www/mib2/vendor/statamic/cms/src/View/Antlers/Parser.php(177): Statamic\\View\\Antlers\\Parser->parseCallbackTags()
#22 /var/www/mib2/vendor/statamic/cms/src/View/Antlers/Parser.php(132): Statamic\\View\\Antlers\\Parser->parse()
#23 /var/www/mib2/vendor/statamic/cms/src/View/Antlers/Engine.php(95): Statamic\\View\\Antlers\\Parser->parseView()
#24 /var/www/mib2/vendor/laravel/framework/src/Illuminate/View/View.php(139): Statamic\\View\\Antlers\\Engine->get()
#25 /var/www/mib2/vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\\View\\View->getContents()
#26 /var/www/mib2/vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\\View\\View->renderContents()
#27 /var/www/mib2/vendor/statamic/cms/src/View/View.php(94): Illuminate\\View\\View->render()
#28 /var/www/mib2/vendor/statamic/cms/src/Http/Responses/DataResponse.php(154): Statamic\\View\\View->render()
#29 /var/www/mib2/vendor/statamic/cms/src/Http/Responses/DataResponse.php(45): Statamic\\Http\\Responses\\DataResponse->contents()
#30 /var/www/mib2/vendor/statamic/cms/src/Structures/Page.php(382): Statamic\\Http\\Responses\\DataResponse->toResponse()
#31 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(802): Statamic\\Structures\\Page->toResponse()
#32 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(789): Illuminate\\Routing\\Router::toResponse()
#33 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Router->prepareResponse()
#34 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}()
#35 /var/www/mib2/vendor/statamic/cms/src/StaticCaching/Middleware/Cache.php(34): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\StaticCaching\\Middleware\\Cache->handle()
#37 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/AuthGuard.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#38 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\AuthGuard->handle()
#39 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/Localize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#40 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\Localize->handle()
#41 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/StacheLock.php(29): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#42 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\StacheLock->handle()
#43 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#44 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#45 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#46 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#47 /var/www/mib2/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#48 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#49 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#50 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#51 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#52 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#53 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#54 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#55 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#56 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#57 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(723): Illuminate\\Pipeline\\Pipeline->then()
#58 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(698): Illuminate\\Routing\\Router->runRouteWithinStack()
#59 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRoute()
#60 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Routing/Router.php(651): Illuminate\\Routing\\Router->dispatchToRoute()
#61 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch()
#62 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#63 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/DisableFloc.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#64 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\DisableFloc->handle()
#65 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/CheckMultisite.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#66 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CheckMultisite->handle()
#67 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/CheckComposerJsonScripts.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#68 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CheckComposerJsonScripts->handle()
#69 /var/www/mib2/vendor/statamic/cms/src/Http/Middleware/PoweredByHeader.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#70 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\PoweredByHeader->handle()
#71 /var/www/mib2/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(60): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#72 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle()
#73 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#74 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#75 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle()
#76 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#77 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#78 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle()
#79 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#80 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#81 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#82 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#83 /var/www/mib2/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#84 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#85 /var/www/mib2/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#86 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#87 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#88 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then()
#89 /var/www/mib2/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#90 /var/www/mib2/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle()
#91 {main}
```

### Versions

Statamic 3.2.33 Pro
Laravel 8.83.1
PHP 8.0.15
doublethreedigital/duplicator 2.1.0

### Installation

Existing Laravel app

**Comment by ryanmitchell** [comment]
Is locations an entries field or multiple select?

**Comment by erikbml** [comment]
> Is locations an entries field or multiple select?

No: "Which has a custom fieldtype Locations (extends Relationship fieldtype), without a max (so you can link a trainer to multiple locations)"

**Comment by ryanmitchell** [comment]
Ok that’s why it’s not working then. You are comparing an array of values to a single value. You need whereJsonContains() which there is an open PR for

**Comment by duncanmcclean** [comment]
I *believe* this should be do-able now, now that the `->whereJsonContains()` query builder method has been released (#4707).

Try this:

```php
$query->whereJsonContains('locations', $location->id);
```

_^ you'll obviously still need a query scope for this - there's no Antlers equivalent yet._
